tdiff_path/array_feedline_paths.py

```diff
@@ -82,236 +82,256 @@
-                # (incident at mismatch point at balun).
-
-                dB_reflected_at_balun = -1*return_loss_dB + cable_loss
-
-                w_reflected_at_balun = 10 ** (dB_reflected_at_balun / 10)
-
-                w_transmitted_at_balun = power_reaching_balun_w - w_reflected_at_balun
-                # print "transmitted w at balun is {}".format(w_transmitted_at_balun)
-                if w_transmitted_at_balun <= 0:
-                    print radar_name, vswr_files_str
-                    print "Antenna: {}".format(k)
-                    print freq
-                    print "WRONG"
-                    print "This would suggest your cable loss model is too lossy."
-                    print "Cable loss = {} db".format(cable_loss)
-                    sys.exit()
-                reflection_dB_at_balun = 10 * math.log((w_reflected_at_balun /
-                                                        power_reaching_balun_w), 10)
-                transmission_dB_at_balun = 10 * math.log(w_transmitted_at_balun /
-                                                         power_reaching_balun_w, 10)
-
-                # ASSUMING we have a symmetrical mismatch point at the balun and transmission
-                # S12 = S21.
-                # power incoming from antenna will have mismatch point and then cable losses.
-                receive_power = transmission_dB_at_balun - cable_loss
-                receive_power = round(receive_power, 5)
-                phase_rad = float(phase) * math.pi / 180.0
-                data.append((freq, receive_power, float(phase), phase_rad))
-            data = np.array(data, dtype=[('freq', 'i4'), ('magnitude', 'f4'),
-                                         ('phase_deg', 'f4'), ('phase_rad', 'f4')])
-
-            data = reflection_to_transmission_phase(data)
-            # now half the phase for single-direction - unwraps within function.
-
-            if len(data) < min_dataset_length:
-                min_dataset_length = len(data)
-
-            if k[0] == 'M':  # in main files.
-                main_data[k] = data
-            elif k[0] == 'I':  # in intf files
-                intf_data[k] = data
-            else:
-                sys.exit('There is an invalid key {}'.format(k))
-
-            hex_dictionary[k] = hex_colors[0]
-            hex_colors.remove(hex_dictionary[k])
-
-    check_frequency_array(main_data, min_dataset_length)
-    check_frequency_array(intf_data, min_dataset_length)
-
-    all_data = main_data.copy()
-    all_data.update(intf_data)
-
-    # for each antenna, get the linear fit and plot the offset from linear fit.
+
+    dtypes_dict = {'freq': 'Freq*', 'vswr': 'VSWR*', 'phase_deg': 'Phase*'}
+    all_data_phase_wrapped = {}
+    raw_data, colour_dictionary, missing_data, data_description = \
+        retrieve_data_from_csv(plot_location + vswr_files_str, data_location, dtypes_dict)
+
+    min_dataset_length = get_min_dataset_length(raw_data)
```

Comment:
yes - the dataset is the shape[0] of the array so this is getting the shortest dataset of all antennas in case we measured on the Rohde and Schwarz with a different number of points. - added a comment.